src/components/shared/Extensions/ActionButton/types.ts

```diff
@@ -0,0 +1,18 @@
+import { MessageDescriptor } from 'react-intl';
+import { ElementType } from 'react';
+import { ButtonProps } from '../Button/types';
+import { ActionTransformFnType } from '~utils/actions';
+
+export interface ActionButtonProps extends ButtonProps {
+  button?: ElementType;
+  confirmText?: any;
+  error: string;
+  onConfirmToggled?: (...args: any[]) => void;
+  onSuccess?: (result: any) => void;
+  onError?: (error: any) => void;
+  submit: string;
+  success: string;
+  text?: MessageDescriptor | string;
+  transform?: ActionTransformFnType;
+  values?: any | (() => any | Promise<any>);
```

Comment:
the same

Reply:
I've added TODO because it's copied from an existing file (I've moved ActionButton to Extensions) and I don't how do types the variables have for now.